### PR TITLE
build: inline external Github Actions to unblock CI

### PR DESCRIPTION
**Pull request by ktmud**
### SUMMARY

Apache changed policy on Github Actions because of [security risks](https://securitylab.github.com/research/github-actions-preventing-pwn-requests). Now official apache repos can only run a list of allowed actions.

CI has been blocked by this error:

> styfle/cancel-workflow-action@0.6.0 and apache-superset/cached-dependencies@b90713b are not allowed to be used in apache/incubator-superset. Actions in this workflow must be: created by GitHub, verified in the GitHub Marketplace, within a repository owned by apache or match the following: apache/*, dawidd6/action-download-artifact@*, gradle/wrapper-validation-action, gradle/wrapper-validation-action@*, peter-evans/create-pull-request@*, scacap/action-surefire-report@*, shivammathur/setup-php@*. 

This PR unblocks CI by
1. Remove `styfle/cancel-workflow-action`, which may soon become obsolete if #11944 gets merged
1. Move all other external dependencies to be within current repo, including:
    - [apache-superset/cached-dependencies](https://github.com/apache-superset/cached-dependencies) or ktmud/cached-dependencies
    - [MorrisonCole/pr-lint-action](https://github.com/MorrisonCole/pr-lint-action)
    - [unsplash/comment-on-pr](https://github.com/unsplash/comment-on-pr/tree/v1.2.0)
    - [trilom/file-changes-action](https://github.com/trilom/file-changes-action/tree/v1.2.4)
    
    These actions were copied over via downloading them as ZIP files directly from GitHub.  All of these actions use the MIT license so I'd assume there is no licensing issue.
1. Change `pull_request_target` trigger for the E2E action to `pull_request`. This is needed for unblocking the CI that runs on existing workflow configs. This will also prevent future Pwn requests and we don't plan to bring it back.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TEST PLAN

CI should pass

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by villebro — APPROVED** [review]
Thanks for taking this on. Minor typo, other than that makes sense and LGTM.

File: .github/actions/cached-dependencies/README.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
+# cached-dependencies
+
+Enable **multi-layer cache** and **shortcut commands** in any workflows.
+
+Manage multiple cache targets in one step. Use either the built-in cache configs for npm, yarn, and pip, or write your own. Create a bash command library to easily reduce redudencies across workflows. Most useful for building webapps that require multi-stage building processes.
```
[the remark this answers is not in the document]
Comment (ktmud) [reply .github/actions/cached-dependencies/README.md:7]: These files were copied over via "Download as ZIP file". I'd imagine future updates for these actions also go with the same route (after more strict security reviews), so I'll leave the typos as is.

[the remark this answers is not in the document]
Comment (villebro) [reply .github/actions/cached-dependencies/README.md:7]: Ah, makes sense, I thought the README was locally generated.

**Comment by villebro** [comment]
I'm going to merge this to unblock the long list of PRs.

**Comment by junlincc** [comment]
massive take on! really appreciate your help! @ktmud 🙏

**Pull request merged**